go/pkg/sysdb/coordinator/table_catalog.go

```diff
@@ -307,26 +310,53 @@ func (tc *Catalog) createCollectionImpl(txCtx context.Context, createCollection
 		LastCompactionTimeSecs:     createCollection.LastCompactionTimeSecs,
 	}
 
-	err = tc.metaDomain.CollectionDb(txCtx).Insert(dbCollection)
+	created := false
+	// In get or create mode, ignore conflicts
+	// NOTE(hammadb) 5/16/2025 - We could skip the above get() and always InsertOnConflictDoNothing
+	// however when adding this change to handle a race condition I am biasing towards keeping the exisitng
+	// performance of the happy get() path without profiling whether InsertOnConflictDoNothing is equivalent
+	// If this proves to be a performance issue, we can change this to always use InsertOnConflictDoNothing
+	// and remove the get() above
+	if createCollection.GetOrCreate {
+		created, err = tc.metaDomain.CollectionDb(txCtx).InsertOnConflictDoNothing(dbCollection)
+	} else {
+		err = tc.metaDomain.CollectionDb(txCtx).Insert(dbCollection)
```

Comment:
can this lead to inconsistency? For e.g.
1. Thread 1 calls create_collection and inserts a row
2. Thread 2 calls delete and deletes the row
3. Thread 3 calls create_collection and creates a row
4. Thread 1 now proceeds to insert metadata and inserts the metadata

Thus collection metadata is of thread 1 whereas collection info is of thread 3

Reply:
Discussed offline that we are ok with timing assumption that hard delete can't happen in (2)